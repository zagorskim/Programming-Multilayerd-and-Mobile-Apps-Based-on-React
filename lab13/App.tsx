import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import { Title } from "react-native-paper";
import CountryList from "./components/CountryList";

export default function App() {
  return (
    <View style={styles.container}>
      <Title style={{ color: "black", fontSize: 50, paddingTop: 100, paddingBottom: 20 }}>
        Country list
      </Title>
      <CountryList />
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "flex-start",
  },
});
