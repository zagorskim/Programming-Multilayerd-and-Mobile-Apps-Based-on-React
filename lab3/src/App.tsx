import { CarsListComponent } from './components/CarsListComponent';
import SearchBar from './components/SearchBar';
 
const App = () => {
    return (
        <div>
            <CarsListComponent/>
        </div>
    );
}

export default App;
