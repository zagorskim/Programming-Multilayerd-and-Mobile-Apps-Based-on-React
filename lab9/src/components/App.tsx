import React from 'react';
import EmployeeList from "./employeeList/EmployeeList";

const App = () => (

    <>
        <EmployeeList/>
    </>
)

export default App;
