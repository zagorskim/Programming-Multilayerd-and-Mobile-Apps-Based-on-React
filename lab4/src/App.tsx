import React from 'react';
import './App.css';
import { CustomerForm } from './components/CustomerForm';


function App ()  {
  return (
    <div className="App">
      <CustomerForm/>
    </div>
  );
}
// why error?
// export default App;

// import React from 'react';
// import './App.css';
// import { CustomerForm } from './components/CustomerForm';

// type AppProps = {
//   counter: number;
// }

// function App: React.FC<AppProps> = (props: AppProps) => {
//   return (
//     <div className="App">
//       <CustomerForm/>
//     </div>
//   );
// }

 export default App;
