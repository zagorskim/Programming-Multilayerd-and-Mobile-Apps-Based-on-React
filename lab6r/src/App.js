import React from 'react';
import { Counter } from './features/counter/Counter';
import {Table} from './features/snake/components/Table'
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <Counter />
        <Table />
      </header>
    </div>
  );
}

export default App;
