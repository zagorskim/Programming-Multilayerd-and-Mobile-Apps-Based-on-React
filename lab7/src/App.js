import logo from './logo.svg';
import './App.css';
import {RecoilRoot} from 'recoil';
import Match from './components/Match.jsx'

export default function App() {
  return (
    <RecoilRoot>
      <div className="App">
        <Match />
      </div>
    </RecoilRoot>
  );
}

