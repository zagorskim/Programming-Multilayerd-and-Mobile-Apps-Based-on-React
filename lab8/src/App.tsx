import React from 'react';
import logo from './logo.svg';
import './App.css';
import { HomePage } from './components/HomePage';
import { AboutPage } from './components/AboutPage';
import { Error404Page } from './components/Error404Page';
import { Routes, Route, Link } from "react-router-dom";
import { Layout } from './components/Layout';
import { CarsPage } from './components/CarsPage';
import { NewCarPage } from './components/NewCarPage';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<Layout />} >
          <Route path="/" element={<HomePage />} />
          <Route path="about" element={<AboutPage />} />
          <Route path="cars" element={<CarsPage />}>
            <Route path='new' element={<NewCarPage/>}/>
          </Route>
          <Route path="*" element={<Error404Page />} />
        </Route>
      </Routes>

    </div>
  );
}

export default App;
