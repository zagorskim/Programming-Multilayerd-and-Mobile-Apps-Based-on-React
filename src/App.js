import './App.css';
import GameAdmin from './GameAdmin.js';

function App() {
  return (
    <div className="App">
      <header>
        <GameAdmin numberOfPlayers = {2} initialName1="name1" initialName2="name2" initialCounter1="0" initialCounter2="0"/>
      </header>
      <body>

      </body>
    </div>
  );
}

export default App;
