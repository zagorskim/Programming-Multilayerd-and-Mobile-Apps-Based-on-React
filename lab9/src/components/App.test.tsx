/* eslint-disable testing-library/no-node-access */
import React from "react";
import {render, screen} from "@testing-library/react";
import App from "./App";

/*
getBy - finds or throws error - for standard use
queryBy - finds or returns null - used to verify that something is not there
findBy - async, finds or throws
*/

//region module mocks
jest.mock("./employeeList/EmployeeList", () => {
    return function EmployeeListMock() {
        return (
            <div>EmployeeListMock</div>
        );
    };
});
//endregion

describe('App component test', () => {
    test('Mocked EmployeeList should be rendered in App', () => {
        render(<App/>);
        expect(screen.getByText('EmployeeListMock')).toBeTruthy();
    });
});

describe('App snapshot tests', () => {
    test('App with mocked List should match inline snapshot', () => {
        const {container} = render(<App/>);
        expect(container.firstChild).toMatchInlineSnapshot(`
    <div>
      EmployeeListMock
    </div>
  `)
    });

    test('App with mocked List should match snapshot', () => {
        const {container} = render(<App/>);
        expect(container.firstChild).toMatchSnapshot();
    })
});