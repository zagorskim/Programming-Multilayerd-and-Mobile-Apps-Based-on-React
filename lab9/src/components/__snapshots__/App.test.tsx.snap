// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot tests App with mocked List should match snapshot 1`] = `
<div>
  EmployeeListMock
</div>
`;
